src/embedding/common.ts

```diff
@@ -224,16 +225,39 @@ export namespace EmbeddedGdc {
     // re-export from AFM namespace
     export type IPositiveAttributeFilter = AFM.IPositiveAttributeFilter;
     export type INegativeAttributeFilter = AFM.INegativeAttributeFilter;
-    export type IAbsoluteDateFilter = AFM.IAbsoluteDateFilter;
-    export type IRelativeDateFilter = AFM.IRelativeDateFilter;
+    export interface IAbsoluteDateFilter {
+        absoluteDateFilter: {
+            // dataSet is required in AD only
+            dataSet?: ObjQualifier;
+            from: string;
+            to: string;
+        };
+    }
+    export interface IRelativeDateFilter {
+        relativeDateFilter: {
+            // dataSet is required in AD only
+            dataSet?: ObjQualifier;
+            granularity: string;
+            from: number;
+            to: number;
+        };
+    }
     export type AttributeFilterItem = IPositiveAttributeFilter | INegativeAttributeFilter;
     export type DateFilterItem = IAbsoluteDateFilter | IRelativeDateFilter;
     export type FilterItem = DateFilterItem | AttributeFilterItem;
     export type ObjQualifier = AFM.ObjQualifier;
 
-    export const isDateFilter = AFM.isDateFilter;
-    export const isRelativeDateFilter = AFM.isRelativeDateFilter;
-    export const isAbsoluteDateFilter = AFM.isAbsoluteDateFilter;
+    type ExtendedFilter = FilterItem | AFM.IMeasureValueFilter;
+    type CompatibilityFilter = AFM.IExpressionFilter | ExtendedFilter;
```

Comment:
should be `type CompatibilityFilter = AFM.IExpressionFilter | FilterItem | AFM.IMeasureValueFilter;`

Reply:
updated